source/server/active_tcp_socket.cc

```diff
@@ -142,7 +200,9 @@ void ActiveTcpSocket::newConnection() {
     }
     // Reset the file events which are registered by listener filter.
     // reference https://github.com/envoyproxy/envoy/issues/8925.
-    socket_->ioHandle().resetFileEvents();
+    if (listener_filter_buffer_ != nullptr) {
+      listener_filter_buffer_->reset();
```

Comment:
Let me double-check this, seems the event should be reset for the rebalancing case at line 195. And we don't have tests to cover this case.